src/lib/include/xtgeo/numerics.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef XTGEO_NUMERICS_HPP_
+#define XTGEO_NUMERICS_HPP_
+
+namespace XTGeo::Numerics {
+
+template<typename T>
+struct Vec3
+{
+    T x, y, z;
+};
+
+inline Vec3<double>
+lerp3d(double x1, double y1, double z1, double x2, double y2, double z2, double t)
```

Comment:
Named `lerp3d` because C++ has a standard library function for linear interpolation, `std::lerp` (which is of course not 3d)